lib/service/moduleGenerator.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright (c) 2016 Kinvey Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+'use strict';
+
+const _ = require('lodash');
+
+const backendContext = require('./modules/backendContext');
+const email = require('./modules/email');
+const entity = require('./modules/entity');
+const kinveyDate = require('./modules/kinveyDate');
+const logger = require('./modules/logger');
+const push = require('./modules/push');
+const requestContext = require('./modules/requestContext');
+const tempObjectStore = require('./modules/tempObjectStore');
+const validation = require('./modules/validation');
+
+const EventEmitter = require('events').EventEmitter;
+
+const proxyTaskEmitter = new EventEmitter();
+let pendingProxyTasks = 0;
+
+proxyTaskEmitter.on('proxyTaskStarted', () => { pendingProxyTasks += 1; });
+proxyTaskEmitter.on('proxyTaskCompleted', () => { pendingProxyTasks -= 1; });
+
+const generateModules = (task) => {
+  const getSecurityContextString = (authorizationHeader, appMetadata) => {
+    const encodedCredentials = authorizationHeader.split(' ')[1];
+    const credentials = new Buffer(encodedCredentials, 'base64').toString('ascii').split(':');
+
+    if (_.isArray(credentials)) {
+      if (credentials[0] !== appMetadata._id) {
+        return 'user';
+      }
+      if (credentials[0] === appMetadata._id) {
+        if (credentials[1] === appMetadata.appsecret) {
```

Comment:
You're checking that `credentials` is an array, but not that it has at least two elements.